cmd/dep/gopath_analyzer.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"github.com/golang/dep"
+	fb "github.com/golang/dep/internal/feedback"
+	"github.com/golang/dep/internal/gps"
+	"github.com/golang/dep/internal/gps/pkgtree"
+)
+
+// gopathAnalyzer deduces configuration from the projects in the GOPATH
+type gopathAnalyzer struct {
+	ctx  *dep.Ctx
+	pkgT pkgtree.PackageTree
+	cpr  string
+	sm   *gps.SourceMgr
+
+	pd    projectData
+	origL *dep.Lock
+}
+
+func newGopathAnalyzer(ctx *dep.Ctx, pkgT pkgtree.PackageTree, cpr string, sm *gps.SourceMgr) *gopathAnalyzer {
+	return &gopathAnalyzer{
+		ctx:  ctx,
+		pkgT: pkgT,
+		cpr:  cpr,
+		sm:   sm,
+	}
+}
+
+// Perform analysis of the filesystem tree rooted at path, with the
+// root import path importRoot, to determine the project's constraints, as
+// indicated by a Manifest and Lock.
+func (a *gopathAnalyzer) DeriveRootManifestAndLock(path string, n gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error) {
+	var err error
+
+	a.pd, err = getProjectData(a.ctx, a.pkgT, a.cpr, a.sm)
```

Comment:
This is for the gopath detection route (see file name: `gopath_analyzer.go`). I had extracted the logic for init into multiple analyzers (because as I was working on it, made it easier for me to think about). Based on your stated preference for one analyzer, I am now working to combine them back into a single struct.